kafka/metrics.go

```diff
@@ -105,9 +120,22 @@ func (h *metricHooks) OnProduceRecordUnbuffered(r *kgo.Record, err error) {
 // OnFetchRecordUnbuffered records the number of fetched messages.
 // https://pkg.go.dev/github.com/twmb/franz-go/pkg/kgo#HookFetchRecordUnbuffered
 func (h *metricHooks) OnFetchRecordUnbuffered(r *kgo.Record, _ bool) {
+	attrs := attributesFromRecord(r)
+
 	h.messageFetched.Add(context.Background(), 1,
-		metric.WithAttributes(attributesFromRecord(r)...),
+		metric.WithAttributes(attrs...),
 	)
+
+	for _, v := range r.Headers {
+		if v.Key == apmqueue.EventTimeKey {
+			if since, err := time.Parse(time.RFC3339, string(v.Value)); err == nil {
```

Comment:
Should we log the error, it might help us down the line if something bad happens.

Reply:
Good call.
We don't have a logger within any of the telemetry packages. So I've recorded a span error.